skypy/pipeline/driver.py

```diff
@@ -103,7 +103,7 @@ def execute(self, configuration, file_format=None, overwrite=False):
 
         # Variables initialised by value don't require function evaluations
         def isfunction(f):
-            return isinstance(f, Mapping) and 'function' in f
+            return isinstance(f, Mapping)
         variables = {k: v for k, v in config.items() if not isfunction(v)}
```

Comment:
```suggestion
        variables = {k: v for k, v in config.items() if not isinstance(v, Mapping)}
```
This is effectively what is happening now. Does this mean that we can't define dictionaries as variables? Is this an issue?

Reply:
This is an issue, which is essentially the question where variables or functions can appear. I think the logical next step from this change would be to have columns that support multiple function calls,
```yaml
my_column:
- my_module.my_sequential_function_1:
    arg1: ...
    arg2: ...
- my_module.my_sequential_function_2:
    arg3: ...
    arg4: ...
```
which would bring the same issue with lists and variables as lists (more common than dicts perhaps). So if there can be better specified rules for where variables can appear (can columns be variables?) that would resolve both situations.

Reply:
So this is strings vs. variable names all over again.

Reply:
We could have a config section `variables` where all entires are assigned their values as parsed by `pyyaml`. That would also conveniently get around the need for escaping literal strings as you could just define them all as variables and then assume any string argument for a function is actually a variable name. Then perhaps have another section for "variables" but that are calculated from functions:
```
cosmology:
    astropy.cosmology.default_cosmology.get: []

variables:
    int_variable: 1
    float_variable: 2.3
    string_variable: A literal string
    list_variable:
    - element1
    - element2
    dict_variable:
        key1: value1
        key2: value2

variables_from_functions: # Obviously needs a better name
    two_squared:
        np.square: [2]
    gamma_one:
        scipy.special.gamma: [int_variable]

tables:
    ....
```

Reply:
I think I found a rather neat solution. Literal strings are tagged with `!str` and literal dicts are tagged with `!dict`. That can be expanded to any other type we want to support in the future.